### ISSUE TITLE
sparse_softmax_cross_entropy_with_logits wrong answer under XLA

### ISSUE THREAD

**Issue by chrism0dwk** [tensorflow/tensorflow#40606]
Related to [https://github.com/tensorflow/probability/issues/975](https://github.com/tensorflow/probability/issues/975)

## System spec

TF version: 2.3.0-dev20200618 (`tf-nightly`)
OS: Linux Mint 19.3, CUDA-10.1.

## Current behaviour

Under XLA compilation, `tf.nn.sparse_softmax_cross_entropy_with_logits ` returns `nan` if one or more `logits` are `-inf`.  In regular graph mode, `-0.0` is returned.

## Expected behaviour
The log_prob function above should return the cross entropy of obtaining labels `k` from logits `logits`, and be non-`nan` (`-inf` is okay) for valid values of the function parameters.  

**NB** expected behaviour occurs under TF2.2.0 (stock Colab).  Only the current dev build is affected. 

## Minimal working example

 [colab here](https://colab.research.google.com/drive/1OUwqDF_KU0aUo-NHq-HKtxUgYv0Szy-u?usp=sharing)
```python
probs = tf.constant([0, 1, 1], dtype=tf.int32)
logits = tf.math.log(probs)

@tf.function(autograph=False, experimental_compile=True)
def log_prob(k):                                                                                   
    lp = -tf.nn.sparse_softmax_cross_entropy_with_logits(labels=k, logits=logits)                                   
    return lp  
print(log_prob(1))  # Expected: -0.6931472, Actual: nan
```

**Comment by ravikyram** [comment]
@chrism0dwk 

I have tried in colab with TF nightly version(`2.3.0-dev20200619`) and was able to reproduce the issue.But i am not seeing nay issue with TF 2.2 version.Please, find the gist [here](https://colab.research.google.com/gist/ravikyram/a216776610471b160d8f9f12e0d5be75/untitled49.ipynb).Thanks!

**Comment by chrism0dwk** [comment]
Yup, confirmed that's it.  Thanks.

**Comment by brianwa84** [comment]
@srvasude could you take a look at this?

**Comment by srvasude** [comment]
This is fixed by https://github.com/tensorflow/tensorflow/commit/e9b3d33ccfa4b670ca0a221db9bc7bc1281b0557
